test/sql/attach/attach_if_not_exists.test

```diff
@@ -0,0 +1,19 @@
+# name: test/sql/attach/attach_if_not_exists.test
+# description: Test attaching of a read-only database
```

Comment:
Can we also add tests like this:
```sql
statement ok
ATTACH '__TEST_DIR__/attach_if_not_exists.db' AS db1 (READ_ONLY);

statement error
ATTACH IF NOT EXISTS '__TEST_DIR__/attach_if_not_exists.db' AS db1;
----
already attached read-only database with the same path, consider making this read-only as well

statement ok
ATTACH '__TEST_DIR__/attach_if_not_exists2.db' AS db2;

statement error
ATTACH IF NOT EXISTS '__TEST_DIR__/attach_if_not_exists2.db' AS db2 (READ_ONLY);
----
already attached database with the same path, consider removing read-only
```
Are there any other flags that might conflict here?

And a small nit: the test description is off here.